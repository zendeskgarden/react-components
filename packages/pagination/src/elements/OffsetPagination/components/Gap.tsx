/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef, LiHTMLAttributes } from 'react';
import { StyledGapListItem } from '../../../styled';
import { useText } from '@zendeskgarden/react-theming';

const GapComponent = forwardRef<HTMLLIElement, LiHTMLAttributes<HTMLLIElement>>((props, ref) => {
  const ariaLabel = useText(
    GapComponent,
    props,
    'aria-label',
    'Ellipsis indicating non-visible pages'
  );

  return (
    <StyledGapListItem {...props} aria-label={ariaLabel} ref={ref}>
      â€¦
    </StyledGapListItem>
  );
});

GapComponent.displayName = 'Pagination.Gap';

/**
 * @extends LiHTMLAttributes<HTMLLIElement>
 */
export const Gap = GapComponent;
