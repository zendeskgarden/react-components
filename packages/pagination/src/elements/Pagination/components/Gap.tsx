/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef, LiHTMLAttributes } from 'react';
import { StyledGap } from '../../../styled';

const GapComponent = forwardRef<HTMLLIElement, LiHTMLAttributes<HTMLLIElement>>((props, ref) => (
  <StyledGap {...props} ref={ref}>
    â€¦
  </StyledGap>
));

GapComponent.displayName = 'Pagination.Gap';

/**
 * @extends LiHTMLAttributes<HTMLLIElement>
 */
export const Gap = GapComponent;
