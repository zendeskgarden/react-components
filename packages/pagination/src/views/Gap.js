import PropTypes from 'prop-types';
import styled from 'styled-components';
import PaginationStyles from '@zendesk/garden-css-pagination';
import { retrieveTheme } from '@zendesk/garden-react-theming';

import { version } from '../../package.json';
import Page from './Page';
const COMPONENT_ID = 'pagination.gap';

/**
 * Accepts all `<li>` props
 */
const Gap = styled(Page).attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  className: PaginationStyles['c-pagination__page--gap']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

Gap.propTypes = {
  current: PropTypes.bool,
  focused: PropTypes.bool,
  hovered: PropTypes.bool,
  hidden: PropTypes.bool
};

/** @component */
export default Gap;
