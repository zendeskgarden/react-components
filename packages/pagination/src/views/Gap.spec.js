/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { render } from 'garden-test-utils';

import Gap from './Gap';

describe('Gap', () => {
  it('renders default styling', () => {
    const { container } = render(<Gap />);

    expect(container.firstChild).toHaveClass('c-pagination__page--gap');
  });
});
