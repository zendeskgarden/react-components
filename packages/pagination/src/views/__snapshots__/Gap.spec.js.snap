// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gap renders default styling 1`] = `
<Page
  className="c-pagination__page--gap "
  data-garden-id="pagination.gap"
  data-garden-version="version"
/>
`;
